mbedtls-sys/build/config.rs

```diff
@@ -362,23 +362,25 @@ pub const DEFAULT_DEFINES: &'static [CDefine] = &[
 
 #[cfg_attr(rustfmt, rustfmt_skip)]
 pub const FEATURE_DEFINES: &'static [(&'static str, CDefine)] = &[
-    ("time",            ("MBEDTLS_HAVE_TIME",             Defined)),
-    ("time",            ("MBEDTLS_HAVE_TIME_DATE",        Defined)),
-    ("time",            ("MBEDTLS_TIMING_C",              Defined)),
-    ("havege",          ("MBEDTLS_HAVEGE_C",              Defined)),
-    ("threading",       ("MBEDTLS_THREADING_C",           Defined)),
-    ("pthread",         ("MBEDTLS_THREADING_PTHREAD",     Defined)),
-    ("custom_threading",("MBEDTLS_THREADING_IMPL",        Defined)),
-    ("pkcs11",          ("MBEDTLS_PKCS11_C",              Defined)),
-    ("zlib",            ("MBEDTLS_ZLIB_SUPPORT",          Defined)),
-    ("std",             ("MBEDTLS_NET_C",                 Defined)),
-    ("std",             ("MBEDTLS_FS_IO",                 Defined)),
-    ("std",             ("MBEDTLS_NO_PLATFORM_ENTROPY",   Undefined)),
-    ("std",             ("MBEDTLS_DEBUG_C",               Defined)),
-    ("std",             ("MBEDTLS_ENTROPY_C",             Defined)),
-    ("custom_printf",   ("MBEDTLS_PLATFORM_C",            Defined)),
-    ("custom_printf",   ("MBEDTLS_PLATFORM_PRINTF_MACRO", DefinedAs("mbedtls_printf"))),
-    ("aesni",           ("MBEDTLS_AESNI_C",               Defined)),
+    ("time",               ("MBEDTLS_HAVE_TIME",               Defined)),
+    ("time",               ("MBEDTLS_HAVE_TIME_DATE",          Defined)),
+    ("time",               ("MBEDTLS_TIMING_C",                Defined)),
+    ("havege",             ("MBEDTLS_HAVEGE_C",                Defined)),
+    ("threading",          ("MBEDTLS_THREADING_C",             Defined)),
+    ("pthread",            ("MBEDTLS_THREADING_PTHREAD",       Defined)),
+    ("custom_threading",   ("MBEDTLS_THREADING_IMPL",          Defined)),
+    ("pkcs11",             ("MBEDTLS_PKCS11_C",                Defined)),
+    ("zlib",               ("MBEDTLS_ZLIB_SUPPORT",            Defined)),
+    ("std",                ("MBEDTLS_NET_C",                   Defined)),
+    ("std",                ("MBEDTLS_FS_IO",                   Defined)),
+    ("std",                ("MBEDTLS_NO_PLATFORM_ENTROPY",     Undefined)),
+    ("std",                ("MBEDTLS_DEBUG_C",                 Defined)),
+    ("std",                ("MBEDTLS_ENTROPY_C",               Defined)),
+    ("custom_printf",      ("MBEDTLS_PLATFORM_C",              Defined)),
+    ("custom_printf",      ("MBEDTLS_PLATFORM_PRINTF_MACRO",   DefinedAs("mbedtls_printf"))),
+    ("aesni",              ("MBEDTLS_AESNI_C",                 Defined)),
+    ("padlock",            ("MBEDTLS_PADLOCK_C",               Defined)),
+    ("custom_has_support", ("MBEDTLS_CUSTOM_HAS_AESNI",        Defined)),
```

Comment:
Also define `MBEDTLS_CUSTOM_HAS_PADLOCK` if `custom_has_support`

Reply:
Just wanted to confirm, with current changes, we do not require a custom has_padlock feature right?
As in we can just enable/disable entire padlock support based on users wish, and in SGX, there wont be a padlock support needed....

Reply:
The `custom_has_support` feature used to enable custom has_support functions for both AES-NI and PadLock. For semver compatibility, any changes we make must keep that behavior.

Reply:
but we did not have a custom_has for pad lock in the past, we made it a weak symbol and let it in.
If we now, want to have a custon_has for pad lock too, we will need to implement one. I am ok with adding one in mbedtls/lib.rs. Let me know your thoughts...